lib/validation-stream.js

```diff
@@ -9,6 +9,12 @@ var Duplex = require('readable-stream').Duplex;
 
 var XmlParser = require('./xml-parser');
 
+/** @arg {?string} str Input string to check.
+    @return {!boolean} True if input is undefined, null, or entirely whitespace. */
```

Comment:
I find that annotating nullity expectations is invaluable but to each their own. I would recommend at least keeping a comment about the return value since different folks might have different expectations over what "blank" means. I used the definition from [Apache Commons' StringUtils](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isBlank-java.lang.CharSequence-).